File: hello/python/greeter_test.py
import greeter
import unittest

class HelloTest( unittest.TestCase ):

    def test_name_not_provided_1( self ):
        self.assertEqual( "Hello!", greeter.hello() )

    def test_name_not_provided_2( self ):
        self.assertEqual( "Hello!", greeter.hello( None ) )

    def test_name_provided( self ):
        self.assertEqual( "Hello, Konstantin!", greeter.hello( "Konstantin" ) )

if __name__ == '__main__':
    unittest.main()

File: hello/ruby/greeter-test.rb
require_relative "greeter"
require "test/unit"
 
class GreeterTest < Test::Unit::TestCase
 
  def test_name_not_provided_1
    assert_equal( "Hello!", hello() )
  end
 
  def test_name_not_provided_2
    assert_equal( "Hello!", hello( nil ) )
  end
 
  def test_name_provided
    assert_equal( "Hello, Konstantin!", hello( "Konstantin" ) )
  end
 
end
